python/lsst/afw/detection/footprint.cc

```diff
@@ -22,16 +22,113 @@
+            "bbox"_a, "region"_a=geom::Box2I());
+    clsFootprint.def(py::init<geom::Point2I const &, double const, geom::Box2I const &>(),
+            "center"_a, "radius"_a, "region"_a=geom::Box2I());
+    clsFootprint.def(py::init<geom::ellipses::Ellipse const &, geom::Box2I const &>(),
+            "ellipse"_a, "region"_a=geom::Box2I());
+    clsFootprint.def(py::init<typename Footprint::SpanList const &, geom::Box2I const &>(),
+            "spans"_a, "region"_a);
+    clsFootprint.def(py::init<Footprint const &>());
+
+    clsFootprint.def("assign", &Footprint::operator=);
+
+    // No good way to wrap the bitmask, not sure if it is needed
+    clsFootprint.def("intersectMask", [](Footprint & self, image::Mask<image::MaskPixel> const & mask) { return self.intersectMask(mask); });
+    clsFootprint.def("isHeavy", &Footprint::isHeavy);
+    clsFootprint.def("getId", &Footprint::getId);
+    clsFootprint.def("getSpans", (typename Footprint::SpanList& (Footprint::*)()) &Footprint::getSpans);
+    clsFootprint.def("getPeaks", (PeakCatalog & (Footprint::*)()) &Footprint::getPeaks, py::return_value_policy::reference_internal);
+    clsFootprint.def("addPeak", &Footprint::addPeak);
+    clsFootprint.def("sortPeaks", &Footprint::sortPeaks,
+            "key"_a=afw::table::Key<float>());
+    clsFootprint.def("setPeakSchema", &Footprint::setPeakSchema);
+    clsFootprint.def("getNpix", &Footprint::getNpix);
+    clsFootprint.def("getArea", &Footprint::getArea);
+    clsFootprint.def("getCentroid", &Footprint::getCentroid);
+    clsFootprint.def("getShape", &Footprint::getShape);
+    clsFootprint.def("addSpan", (const Span& (Footprint::*)(const int, const int, const int)) &Footprint::addSpan);
+    clsFootprint.def("addSpan", (const Span& (Footprint::*)(Span const &)) &Footprint::addSpan);
+    clsFootprint.def("addSpan", (const Span& (Footprint::*)(Span const &, int, int)) &Footprint::addSpan);
+    clsFootprint.def("addSpanInSeries", &Footprint::addSpanInSeries);
+    clsFootprint.def("shift", (void (Footprint::*)(int, int)) &Footprint::shift);
+    clsFootprint.def("shift", (void (Footprint::*)(geom::ExtentI)) &Footprint::shift);
+    clsFootprint.def("getBBox", &Footprint::getBBox);
+    clsFootprint.def("getRegion", &Footprint::getRegion);
+    clsFootprint.def("setRegion", &Footprint::setRegion);
+    clsFootprint.def("clipTo", &Footprint::clipTo);
+    clsFootprint.def("contains", &Footprint::contains);
+    clsFootprint.def("normalize", &Footprint::normalize);
+    clsFootprint.def("isNormalized", &Footprint::isNormalized);
+    clsFootprint.def("transform", &Footprint::transform,
+            "source"_a, "target"_a, "region"_a, "doClip"_a=true);
+    clsFootprint.def("findEdgePixels", &Footprint::findEdgePixels);
+    clsFootprint.def("include", &Footprint::include,
+            "children"_a, "ignore"_a=false);
+    clsFootprint.def("isPersistable", &Footprint::isPersistable);
+
+    declareMask<int>(clsFootprint);
```

Comment:
In general I'm all for specific integer types but in this case it has to match the instantiations in the `.cc` files.